src/main/java/org/folio/service/pieces/flows/delete/PieceDeleteFlowManager.java

```diff
@@ -79,4 +85,26 @@ protected Future<Void> updatePoLine(PieceDeletionHolder holder, RequestContext r
       : pieceDeleteFlowPoLineService.updatePoLine(holder, requestContext);
   }
 
+  public Future<List<Void>> batchDeletePiece (PieceCollection entity, RequestContext requestContext) {
+    List <String> ids = new ArrayList<>();
+    entity.getPieces().stream().forEach(v -> ids.add(v.getId()));
+    List<Future<PieceDeletionHolder>> deletionHolders = ids.stream()
+      .map(pieceId -> {
+        PieceDeletionHolder holder = new PieceDeletionHolder().withDeleteHolding(true);
```

Comment:
we could not hardcode deleteHolding to true, it should be fetched from request param as doing in Delete By Id endpoint

Reply:
thank you for pointing it out, I will correct.